src/control/server/ctl_storage_rpc.go

```diff
@@ -151,6 +159,112 @@ func (c *ControlService) scanScm(ctx context.Context, req *ctlpb.ScanScmReq) (*c
 	return newScanScmResp(c.getScmUsage(ssr))
 }
 
+// Return the number of target managing a given NVME controleur
+func (c *ControlService) getTargetCount(pciAddr string) (int, error) {
+	for _, engineCfg := range c.srvCfg.Engines {
+		targetCount := engineCfg.TargetCount
+		for _, tierCfg := range engineCfg.Storage.Tiers {
+			if tierCfg.IsBdev() {
+				for index := range tierCfg.Bdev.DeviceList {
+					if pciAddr == tierCfg.Bdev.DeviceList[index] {
+						msg := "NVME controler %s is used by %d targets"
+						c.log.Debugf(msg, pciAddr, targetCount)
```

Comment:
This doesn't make sense, targets will be split across different controllers (see `dmg storage scan --nvme-meta` command output) but from what you are doing here your messages indicate that the entire target count is used on each controller.

Reply:
From what I understand, I have a big issue on the algorithm itself.
I have to admit, that I did this part on empirical assumption (i.e. try and test).
I am going to look at the storage request to find the valid number of targets sharing a given nvme controler.
Thanks a lot for this feedback.